signer/src/stacks/api.rs

```diff
@@ -1196,6 +1261,33 @@ impl StacksInteract for StacksClient {
         }
     }
 
+    async fn is_withdrawal_completed(
+        &self,
+        deployer: &StacksAddress,
+        request_id: u64,
+    ) -> Result<bool, Error> {
+        // Both ContractName::from and ClarityName::from can panic when
+        // given the "wrong" strings. These particular strings do not
+        // panic, and we test this fact in our unit tests.
+        let contract_name = ContractName::from("sbtc-registry");
+        let map_name = ClarityName::from("withdrawal-status");
+
+        let map_entry = Value::UInt(request_id as u128);
+        let result = self
+            .get_map_entry(deployer, &contract_name, &map_name, &map_entry)
+            .await?;
```

Comment:
nit: this is fine, I just want to point out that there's also `get-withdrawal-request` that include the status if we don't want to pry into the data maps.

Reply:
Yeah, that reads more data though so I opted to use the map. My impression is that Stacks nodes can have high resource usage or DB contention or something. We'll eventually want to use that read-only function though, maybe I should have added it here.